com.unity.render-pipelines.universal/Runtime/2D/Shadows/ShadowCaster2D.cs

```diff
@@ -75,6 +85,17 @@ static int[] SetDefaultSortingLayers()
             return allLayers;
         }
 
+        internal bool IsLit(Light2D light)
+        {
+            Vector3 deltaPos = light.transform.position - (m_ProjectedBoundingSphere.position + transform.position);
+            float distanceSq = Vector3.SqrMagnitude(deltaPos);
+
+            float lightRadiusSq = light.boundingSphere.radius * light.boundingSphere.radius;
+            float projectedRadiusSq = m_ProjectedBoundingSphere.radius * m_ProjectedBoundingSphere.radius;
+
+            return distanceSq <= (lightRadiusSq + projectedRadiusSq);
```

Comment:
Is this correct? I guess `a^2 <= b^2 + c^2` is not the same as `a <= b+c`

Reply:
Good catch